amplify/backend/function/putUser/src/index.py

```diff
@@ -17,21 +17,25 @@ def lambda_handler(event, context):
     query = f"g.V().has('username', '{username}').id();"
     vertexID = db.submit(query).all().result()
 
-    for attribute in userData:
-        newValue = userData[attribute]
-        # Add new value
-        query = f"g.V({vertexID}).property(single, '{attribute}', '{newValue}');"
-        try:
-            if len(vertexID) == 0:
-                result = f"Server failed to find user : {username}"
-                statusCode = 404
-            else:
-                db.submit(query)
-                result = f"User attribute : \"{attribute}\" successfully changed to : \"{newValue}\"\n"
-                statusCode = 200
-        except:
-            result = f"Server failed at modifying attribute : \"{attribute}\" for user : \"{username}\"\n"
-            statusCode = 500
+    if len(userData) > 0:
+        for attribute in userData:
+            newValue = userData[attribute]
+            # Add new value
+            query = f"g.V({vertexID}).property(single, '{attribute}', '{newValue}');"
+            try:
+                if len(vertexID) == 0:
+                    result = f"Server failed to find user : {username}"
+                    statusCode = 404
+                else:
+                    db.submit(query)
+                    result = f"User attribute : \"{attribute}\" successfully changed to : \"{newValue}\""
+                    statusCode = 200
+            except:
+                result = f"Server failed at modifying attribute : \"{attribute}\" for user : \"{username}\""
+                statusCode = 500
```

Comment:
Minor bug in this logic. If you try to modify a user attribute that doesn't exist, followed by a user attribute that does exists then you will only receive a success indication.

For example:
![image](https://github.com/jackfales/yum/assets/88995035/ac92aec8-77a7-49a4-aa59-580e7c3f17e5)

I don't know if this is too much of an issue as you shouldn't be able to change the attributes being submitted unless you are sending a direct request via Postman, Curl, etc.

Reply:
This is an issue. Our API will ultimately be public, meaning a user should be able to modify their account programatically.

Reply:
Shoot I had it working earlier I just messed it up. Usually it's supposed to append all of the success or failure indications to a list and return that. Will revert back to what I had.